deposit-messaging/src/test/java/org/dataconservancy/pass/deposit/messaging/service/AbstractSubmissionIT.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2018 Johns Hopkins University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.dataconservancy.pass.deposit.messaging.service;
+
+import org.dataconservancy.nihms.builder.fs.PassJsonFedoraAdapter;
+import org.dataconservancy.pass.client.PassClient;
+import org.dataconservancy.pass.model.PassEntity;
+import org.dataconservancy.pass.model.Repository;
+import org.dataconservancy.pass.model.Submission;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.util.HashMap;
+
+import static org.dataconservancy.pass.deposit.messaging.service.SubmissionTestUtil.getDepositUris;
+import static org.dataconservancy.pass.model.Submission.AggregatedDepositStatus.NOT_STARTED;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Elliot Metsger (emetsger@jhu.edu)
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(properties = { "spring.jms.listener.auto-startup=false" })
+public abstract class AbstractSubmissionIT {
+
+    protected static final String EXPECTED_REPO_NAME = "JScholarship";
+
+    protected Submission submission;
+
+    @Autowired
+    @Qualifier("submissionProcessor")
+    protected SubmissionProcessor underTest;
+
+    @Autowired
+    protected PassClient passClient;
+
+    /**
+     * Populates Fedora with a Submission, as if it was submitted interactively by a user of the PASS UI.
+     *
+     * @throws Exception
+     */
+    @Before
+    public void createSubmission() throws Exception {
+        PassJsonFedoraAdapter passAdapter = new PassJsonFedoraAdapter();
+
+        // Upload sample data to Fedora repository to get its Submission URI.
+        InputStream is = getSubmissionResources();
+
+        HashMap<URI, PassEntity> uriMap = new HashMap<>();
+        URI submissionUri = passAdapter.jsonToFcrepo(is, uriMap);
+        is.close();
+
+        // Find the Submission entity that was uploaded
```

Comment:
This can be done with a single line of code (might need a tad of protection).  I offered this same comment on a previous incarnation of this block, so I assume you want to do it this way, but am mentioning it again just in case you didn't see it.

    submission = (Submission)uriMap.get(submissionUri);

Reply:
@Trumbore Hi! I must have flushed your earlier comment from cache, at the expense of my efficiency.

I think the issue was (is? please correct me if I'm wrong!) that the `uriMap` does not have the most recent version of the resource (but I may be mistaken).

For example:

```java
Submission s = new Submission();
// set all the state you want on `Submission`
s.set(...); 
s.set(...);

// Save the resource in the respository
URI submissionUri = passClient.createResource(s);

// assertion fails, because this is state that is set upon creating/saving the 
// Submission, but we don't have that state locally
assertNotNull(s.getVersionTag());

// get the most recent version of the Submission, which includes the state
// set upon saving the resource

s = passClient.readResource(submissionUri, Submission.class);

// assertion passes
assertNotNull(s.getVersionTag());
```

The `PassAdapter` may already get the most recent copy, and I just have my brain hard-wired!  I'm just being extra defensive.

By the way, this issue was the motivation for adding: `PassClient.createAndReadResource(...)`, so the example code above could read:

```java
Submission s = new Submission();
// set all the state you want on `Submission`
s.set(...); 
s.set(...);

// Save the resource in the respository and read it right back
s = passClient.createAndReadResource(s, Submission.class);

// assertion passes
assertNotNull(s.getVersionTag());
```